docs/dev_guide/compare-prog-models.rst

```diff
@@ -0,0 +1,767 @@
+  a work-item instead of a work-group. Thus `global size` should be the product
+  of ``size_1`` and ``size_2`` to align with CUDA.
+* The Thread Indexing section describes that the CUDA execution space right-most
+  dimension *z* should be mapped to the SYCL left-most dimension *x*. Thus in
+  this example, the size of dimension needs to be reversed.
+
+Memory Model
+------------
+
+The CUDA and SYCL memory model is a hierarchical structure. In CUDA, it comprises
+multiple memory spaces, such as shared memory, global memory, constant memory, and
+unified memory. Shared memory allows for efficient communication within a thread
+block. Global memory is accessible to all threads across the device, providing a
+larger storage space but slower access compared to shared memory. Constant memory
+is a read-only space, suitable for storing unchanging data such as constants or
+lookup tables. Unified memory can be accessed both on host and device.
+
+Similarly, in SYCL, local memory is shared within a work-group, global memory is
+accessible by all work-items, and shared memory is accessible on the host and
+device. According to the `SYCL 2020 Specification <https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html>`_, constant memory no longer appears in the SYCL device memory
+model in SYCL 2020.
+
+Shared Memory
+*************
+
+CUDA shared memory can be mapped to SYCL local memory. To perform this migration,
+declare an accessor with access target set to ``sycl::access::target::local``.
+For example:
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Original CUDA Code
+     - Migrated SYCL Code
+   * - .. code-block::
+
+          __global__ void foo() {
+            __shared__ int shm[16];
+            shm[0] = 2;
+          }
+
+          int main() {
+            foo<<<1, 1>>>();
+          }
+
+          void foo(int *shm) {
+           shm[0] = 2;
+          }
+
+     - .. code-block::
+
+          int main() {
+            sycl::queue q;
+            q.submit([&](sycl::handler &cgh) {
+
+              sycl::local_accessor<int> shm_acc(sycl::range<1>(16), cgh);
+              cgh.parallel_for(
+                  sycl::nd_range<3>(sycl::range<3>(1, 1, 1), sycl::range<3>(1, 1, 1)), [=](sycl::nd_item<3> item_ct1) {
+                    foo(shm_acc.get_pointer());
+                  });
+            });
+          }
+
+Global, Constant, and Unified Memory
+************************************
+
+CUDA global memory and constant memory can be mapped to SYCL global memory. CUDA
+unified memory can be mapped to SYCL shared memory. To perform this migration,
+allocate memory through ``sycl::malloc_device`` or ``sycl::malloc_shared``. For
+example:
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Original CUDA Code
+     - Migrated SYCL Code
+   * - .. code-block::
+
+          void foo() {
+            int *mem1, *mem2;
+
+
+            cudaMalloc(&mem1, 10);
+            cudaMallocManaged(*mem2, 10);

```

Comment:
line 309 should be cudaMallocManaged(&mem2, 10);

Reply:
Done.